extractor/edge_based_graph_factory.cpp

```diff
@@ -78,6 +78,11 @@ void EdgeBasedGraphFactory::GetEdgeBasedNodes(std::vector<EdgeBasedNode> &nodes)
     nodes.swap(m_edge_based_node_list);
 }
 
+void EdgeBasedGraphFactory::GetStartPointMarkers(std::vector<bool> &node_is_startpoint)
+{
+    m_edge_based_node_is_startpoint.swap(node_is_startpoint);
+}
```

Comment:
Effective C++, Item 25 (https://books.google.de/books?id=U7lTySXdFk0C&pg=PT139&lpg=PT139&dq=prefer+std::swap+over+member&source=bl&ots=nO6Jrdv9kb&sig=nX3EPlZUTHahOvHTCCSoSba8RmM&hl=en&sa=X&ved=0ahUKEwjCzfvp_trJAhVE_Q4KHRZpA4YQ6AEIOjAF#v=onepage&q=prefer%20std%3A%3Aswap%20over%20member&f=false).

You should consider adding `using std::swap` and call `swap(node_is_starpoint, m_edge_base_node_is_startpoint)` as a general way to invoke swap.